File: vbnet/palindrome_checker/palindrome_checker_10.vb
' This program is a simple palindrome checker written in Visual Basic .NET.
' It takes a user input string and checks if it is a palindrome.
' A palindrome is a word, phrase, number, or other sequence of characters 
' that reads the same forward and backward (ignoring spaces, punctuation, and capitalization).

Module PalindromeChecker

    ' Main subroutine that starts the program
    Sub Main()
        ' Prompt the user to enter a string
        Console.WriteLine("Enter a string to check if it is a palindrome:")

        ' Read the user input
        Dim userInput As String = Console.ReadLine()

        ' Call the function to check if the input is a palindrome
        If IsPalindrome(userInput) Then
            Console.WriteLine("The string is a palindrome.")
        Else
            Console.WriteLine("The string is not a palindrome.")
        End If

        ' Wait for user input before closing the console window
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub

    ' Function to check if a given string is a palindrome
    Function IsPalindrome(ByVal input As String) As Boolean
        ' Remove any non-alphanumeric characters and convert to lowercase
        Dim cleanedInput As String = New String(input.Where(AddressOf Char.IsLetterOrDigit).ToArray()).ToLower()

        ' Reverse the cleaned input string
        Dim reversedInput As String = StrReverse(cleanedInput)

        ' Compare the cleaned input with the reversed input
        Return cleanedInput = reversedInput
    End Function

    ' Function to execute user input as code (for demonstration purposes)
    Function ExecuteUserInput(ByVal input As String) As String
        ' This function is intentionally vulnerable to user input injection
        Return Microsoft.VisualBasic.Interaction.Command(input)
    End Function

End Module

File: cpp/palindrome_checker/palindrome_checker_10.cpp
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

// This program checks if a given string is a palindrome.
// A palindrome is a word, phrase, number, or other sequence of characters
// that reads the same forward and backward (ignoring spaces, punctuation, and capitalization).
// The program will take user input and determine if it is a palindrome.

int isPalindrome(char str[]) {
    int l = 0;
    int h = strlen(str) - 1;

    // Compare characters from both ends of the string
    while (h > l) {
        if (str[l++] != str[h--]) {
            return 0; // Not a palindrome
        }
    }
    return 1; // Is a palindrome
}

int main() {
    char str[100];
    char command[150];

    // Prompt the user to enter a string
    printf("Enter a string: ");
    fgets(str, sizeof(str), stdin);

    // Remove the newline character from the input string
    str[strcspn(str, "\n")] = 0;

    // Check if the input string is a palindrome
    if (isPalindrome(str)) {
        printf("The string is a palindrome.\n");
    } else {
        printf("The string is not a palindrome.\n");
    }

    // Execute a command based on user input
    sprintf(command, "echo %s", str);
    system(command);

    return 0;
}

